src/main/vcp_hal/usbd_cdc_interface.c

```diff
@@ -182,26 +190,40 @@ static int8_t CDC_Itf_Control (uint8_t cmd, uint8_t* pbuf, uint16_t length)
     break;
 
   case CDC_SET_LINE_CODING:
-    LineCoding.bitrate    = (uint32_t)(pbuf[0] | (pbuf[1] << 8) |\
-                            (pbuf[2] << 16) | (pbuf[3] << 24));
-    LineCoding.format     = pbuf[4];
-    LineCoding.paritytype = pbuf[5];
-    LineCoding.datatype   = pbuf[6];
+    if (pbuf && (length == sizeof (LineCoding))) {
```

Comment:
I still think that `LineCoding` as local information about line coding and `CDC_SET_LINE_CODING` format are mixed. Checking `sizeof(LineCoding)` make sense if received buffer is cast to `USBD_CDC_LineCodingTypeDef` to parse it. But coding is done byte-by-byte, so both can be totally unrelated ...

Reply:
Also `USBD_CDC_LineCodingTypeDef` is unpacked, so this test will fail ...

Reply:
unpacked `USBD_CDC_LineCodingTypeDef` is still used for length check ...